react/IOMessage.tsx

```diff
@@ -14,11 +26,39 @@ const IOMessage: IOMessageType = ({ children, id, ...props }) => {
     )
   }
 
+  /**
+   * This enables a user to pass a translatable message which is not
+   * in the current IntlContext to the function and actually have it
+   * formatted by react-intl.
+   */
+  const cache = createIntlCache()
+  const newIntl = createIntl(
+    {
+      locale: intl.locale,
+      messages: {
+        [id]: id,
+      },
+    },
+    cache
+  )
+
   if (children && typeof children === 'function') {
-    return children(intlMessage === '' ? '' : id) as ReactElement || null
+    return (
+      (children(
+        intlMessage === ''
+          ? ''
+          : newIntl.formatMessage({ id, defaultMessage }, props.values)
+      ) as ReactElement) || null
+    )
   }
 
-  return intlMessage === '' ? null : <>{id}</>
+  return intlMessage === '' ? null : (
```

Comment:
I think it would be nice to add comments here explaining when the code can arrive here and what happens when it does so.